### ISSUE TITLE
onXXX or handleXXX callbacks with arguments?

### ISSUE THREAD

**Issue by iocat** [idyll-lang/idyll#299]
Hi! Is there a way to declare a call back on custom React component which also allows argument. 

The current callback onXXX or handleXXX only takes in expression for a non-argument callback function.

**Comment by mathisonian** [comment]
There’s not an explicit way to opt-out at the moment, but there are a few strategies.

But - only expressions (things in backticks) are handled this way. If you pass in a string, number, or bare variable, it won’t be treated as a callback.

Eg `[Comp onX:"string value" /]`

If you have an example use case I’m happy to provide more specific feedback

**Comment by mathisonian** [comment]
So there isn’t a way to do this right now, without writing a custom react component, but this seems like a good thing to support. One way to do it would be to allow people to pass in a function with arguments explicitly, alternatively we could give access to the JavaScrip  arguments object.

Could you comment a little more on your use case?

**Comment by iocat** [comment]
Hi! I was able to solve it, by using a derived callback variable. But I run into another problem. Using the code below.

```idyll

[var name:"x" value:10 /]
[derived name:"changeX" value:\`val=>{x=val; console.log(x)}\`/]
[Select value:x onSelect:changeX]
    [Select.Option value:10] value 10 [/Select.Option]
    [Select.Option value:20] value 20 [/Select.Option]
    [Select.Option value:100] value 100 [/Select.Option]
[/Select]
```

when changeX callback is invoked, the value x is updated, but Select doesn't receive a new x value. Isn't this supposedly an expected behavior?

**Comment by mathisonian** [comment]
Yes - I would consider that a bug. 

You can get similar behavior like

```
[var name:"x" value: 10 /]
[Select value:x options:`[
  { value:10 , label: "value 10" }, 
  { value:20, label: "value 20" },
  { value:100, label: "value 100" },
]` /]
```

... not sure if that helps at all as a workaround.

**Comment by mathisonian** [comment]
Yeah, unfortunately. Though I should be able to get a fix for that issue out relatively soon.

**Comment by mathisonian** [comment]
This is fixed by #367 and available on npm in `idyll@3.2.0`
